enable/trait_defs/kiva_font_trait.py

```diff
@@ -0,0 +1,131 @@
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+""" Trait definition for a wxPython-based Kiva font.
+"""
+
+from pyface.font import Font as PyfaceFont
+from traits.api import DefaultValue, TraitError, TraitType, NoDefaultSpecified
+
+import kiva.constants as kc
+from kiva.fonttools.font import Font, simple_parser
+
+
+font_attrs = [
+    'face_name', 'size', 'family', 'weight', 'style', 'underline', 'encoding',
+]
+
+pyface_family_to_kiva_family = {
+    'default': kc.DEFAULT,
+    'fantasy': kc.DECORATIVE,
+    'decorative': kc.DECORATIVE,
+    'serif': kc.ROMAN,
+    'roman': kc.ROMAN,
+    'cursive': kc.SCRIPT,
+    'script': kc.SCRIPT,
+    'sans-serif': kc.SWISS,
+    'swiss': kc.SWISS,
+    'monospace': kc.MODERN,
+    'modern': kc.MODERN,
+    'typewriter': kc.TELETYPE,
+    'teletype': kc.TELETYPE,
+}
+
+
+def pyface_font_to_font(font):
+    """Convert a Pyface font to an equivalent Kiva Font.
+
+    This ignores stretch and some options like small caps and strikethrough
+    as the Kiva font object can't represent these at the moment.
+
+    Parameters
+    ----------
+    font : Pyface Font instance
+        The font to convert.
+
+    Returns
+    -------
+    font : Kiva Font instance
+        The resulting Kiva Font object.
+    """
+    face_name = font.family[0]
+    for face in font.family:
+        if face in pyface_family_to_kiva_family:
+            family = pyface_family_to_kiva_family[face]
+            break
+    else:
+        family = kc.DEFAULT
+    size = int(font.size)
+    weight = font.weight_
+    style = kc.NORMAL if font.style is 'normal' else kc.ITALIC
+    underline = 'underline' in font.decorations
+    return Font(face_name, size, family, weight, style, underline)
+
+
+class KivaFont(TraitType):
+    """ A Trait which casts strings to a Kiva Font value.
+    """
+
+    #: The default value should be a tuple (factory, args, kwargs)
+    default_value_type = DefaultValue.callable_and_args
+
+    #: The parser to use when converting text to keyword args.  This should
+    #: accept a string and return a dictionary of Font class trait values (ie.
+    #: "family", "size", "weight", etc.).
+    parser = None
+
+    def __init__(self, default_value=None, *, parser=simple_parser, **metadata):
+        self.parser = parser
+        default_value = self._get_default_value(default_value)
+        super().__init__(default_value, **metadata)
+
+    def validate(self, object, name, value):
+        if isinstance(value, Font):
+            return value
+        if isinstance(value, PyfaceFont):
+            return pyface_font_to_font(value)
+        if isinstance(value, str):
+            try:
+                return Font(**self.parser(value))
+            except FontParseError:
+                self.error(object, name, value)
```

Comment:
`FontParseError` is not defined in this scope

Reply:
I guess that also means that the test suite doesn't exercise this branch. (Either that, or the `NameError` is getting swallowed somewhere.) That may be worth fixing.

Reply:
The failure wasn't being discovered because the tests weren't in a package; that's been resolved and the tests should be working now.

There are a few other places where the tests aren't in packages - I suspect that this was a problem introduced when we moved from nose to unittest discover.